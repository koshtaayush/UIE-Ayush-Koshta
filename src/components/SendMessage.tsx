import React from 'react';
import styled from 'styled-components';

import Input from './../shared/Input';
import Button from './../shared/Button';

import { makePost } from './../services/api.service';
import { POST_MESSAGE_API_ENDPOINT } from './../constants/api.constants';
import { SENDER_NAME } from './../config/config';
import { IMessage } from './../typings/sharedInterface';

interface Props {
    addCurrentMessages: (msg: IMessage) => void
}

const SendMessage: React.FC<Props> = (props) => {

    //Props
    const { addCurrentMessages } = props;

    //Current message value which user is typing
    const [messageValue, setMessageValue]=  React.useState<string>('');

    /**
     * Function which is called to handle message change from input box
     */
    const handleMessageChange = (changedMessage: string) => {
        setMessageValue(changedMessage)
    }

    /**
     * Function which is called to submit message
     */
    const handlePostMessage = () => {
        
        const data = {
            message: messageValue,
            author: SENDER_NAME
        }

        makePost(POST_MESSAGE_API_ENDPOINT, data)
            .then((res) => res.json())
            .then((resp) => {
                addCurrentMessages(resp);
                setMessageValue('')
            }, (err) => {
                console.log("Error", err);
            }).catch((error) => {
                console.log("Error", error);
            })

    }

    return (
        <React.Fragment>
            <SendMessageContainer>
                <InputSection>
                    <Input 
                        onChangeProp={(v) => handleMessageChange(v)}
                        inputValue={messageValue}
                        placeHolderText={'Message'}></Input>
                </InputSection>

                <ButtonSection>
                    <Button 
                        buttonType="S"
                        onClick={() => handlePostMessage()}>
                    </Button>
                </ButtonSection>
            </SendMessageContainer>
        </React.Fragment>
    )
}

export default SendMessage

const SendMessageContainer = styled.div`
    background-color: #3798D4;
    width: inherit;
    display: flex;
    align-items: center;
    justify-content: center;
    position: fixed;
    bottom: 0rem;
    max-width: 40rem;
`

const InputSection  = styled.div`
    width: 80%;
    padding: 0.5rem;
`

const ButtonSection = styled.div`
    width: 20%;
    margin-left: 0.5rem;
`