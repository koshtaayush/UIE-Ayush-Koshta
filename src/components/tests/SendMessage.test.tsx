import React from 'react'
import { shallow } from 'enzyme'

import SendMessage from './../SendMessage';

describe('SendMessage', () => {

    it('Should match the snapshot', () => {

        const mockFn = jest.fn()

        const component = shallow(<SendMessage addCurrentMessages={mockFn} />);
        expect(component).toMatchSnapshot();

    })

    it('Should always render input box and Send button', () => {

        const mockFn = jest.fn()
        
        const component = shallow(<SendMessage addCurrentMessages={mockFn} />);  
        expect(component.find({ 'test-id': 'inputSection' })).toHaveLength(1);
        expect(component.find({ 'test-id': 'buttonSection' })).toHaveLength(1);
    })
    
    it('Should render proper text for send button', () => {

        const mockFn = jest.fn()
        
        const component = shallow(<SendMessage addCurrentMessages={mockFn} />);  
        expect(component.find({ 'test-id': 'buttonSection' }).text()).toEqual('Send')
    })
    
})
