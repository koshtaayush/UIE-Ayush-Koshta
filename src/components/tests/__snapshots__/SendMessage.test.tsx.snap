// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendMessage Should match the snapshot 1`] = `
<Fragment>
  <styled.div>
    <styled.div
      test-id="inputSection"
    >
      <Input
        inputValue=""
        onChangeProp={[Function]}
        placeHolderText="Message"
      />
    </styled.div>
    <styled.div
      test-id="buttonSection"
    >
      <Button
        buttonType="S"
        onClickProp={[Function]}
      />
    </styled.div>
  </styled.div>
</Fragment>
`;
